lang/src/org/partiql/lang/planner/PassResult.kt

```diff
@@ -0,0 +1,15 @@
+package org.partiql.lang.planner
+import org.partiql.lang.errors.Problem
+
+sealed class PassResult<TResult> {
+    /**
+     * Indicates query planning was successful and includes a list of any warnings that were encountered along the way.
+     */
+    data class Success<TResult>(val result: TResult, val warnings: List<Problem>) : PassResult<TResult>()
+
+    /**
+     * Indicates query planning was not successful and includes a list of errors and warnings that were encountered
```

Comment:
`errors and warnings` -> `errors`?

Reply:
This is designed to allow us to continue at least the current pass in the event of a semantic error instead of throwing an exception.  That means its entirely possible to encounter both warnings and errors.